src/axom/multimat/multimat.hpp

```diff
@@ -267,21 +305,52 @@ class MultiMat
    * \param (optional) ncomp The number of component of the field. Default is 1
    * \return int the index of the field, can be used to retrieve the field later
    */
-  template <class T>
+  template <typename T>
   int addField(const std::string& field_name,
                FieldMapping field_mapping,
                DataLayout data_layout,
                SparsityLayout sparsity_layout,
-               T* data_array,
+               axom::ArrayView<T> data_array,
                int ncomp = 1);
 
+  /**
+   * \brief Add an externally-managed field to the MultiMat object
+   *
+   * \tparam T The data type (double, float...) of the field
+   * \param field_name The name of the field, used to retrieve the field later
+   * \param field_mapping
+   * \param data_layout
+   * \param sparsity_layout
+   * \param data_array The array containing data to the field. The length of the
+   *            array should be `num_mats * num_cells * ncomp` if the current
+   *            format is Dense, or `num_nonzero * ncomp` if the current format
+   *            is Sprase.
+   * \param (optional) ncomp The number of component of the field. Default is 1
+   * \return int the index of the field, can be used to retrieve the field later
+   */
```

Comment:
Please document what happens if a user tries to add a field and `field_name` already exists.

Reply:
I do realize that adding to the documentation expands the scope of the PR (since the Doxygen for (new) line 309 needs to match this comment).  But `addExternalField` is new work, and the behavior from (new) line 948-954 definitely needs to be documented.

Reply:
I've added documentation for the special-case behavior when adding a volume fraction field or an already-existing field.